app/src/main/java/com/pajato/android/gamechat/exp/model/ChessHelper.java

```diff
@@ -0,0 +1,456 @@
+package com.pajato.android.gamechat.exp.model;
+
+import com.pajato.android.gamechat.R;
+import com.pajato.android.gamechat.exp.ChessPiece;
+import com.pajato.android.gamechat.exp.ChessPiece.ChessTeam;
+
+import java.util.ArrayList;
+
+import static com.pajato.android.gamechat.exp.ChessPiece.PieceType.ROOK;
+
+/**
+ * Helper class to determine for chess features.
+ */
+
+public class ChessHelper {
```

Comment:
I would be hard pressed to call this helper class inspirational.  Apparently it does the job.  Good enough.